src/AppBundle/Service/ImportEntities.php

```diff
@@ -201,171 +240,195 @@ public function verifCsv($softFile, $fileName)
+                                }
+                            } elseif ($property === "list-tag") {
+                                $tags = explode("#", $convertedData[$caseImport]);
+                                foreach ($tags as $tag) {
+                                    $currentTag = $this->em->getRepository(Tag::class)->findOneBy(['name' => $tag,]);
+                                    if (!empty($currentTag)) {
+                                        $eachEntity[$i]->addTag($currentTag);
+                                        $currentTag->addSoftMain($eachEntity[$i]);
+                                        $this->em->persist($currentTag);
+                                    }
                                 }
-                            }
 
-                        } else {
-                            $eachSetter = "set" . ucfirst($listFields[$j]);
-                            $eachEntity[$i]->$eachSetter($convertedData[$caseImport]);
+                            } else {
+                                $eachSetter = "set" . ucfirst($listFields[$j]);
+                                $eachEntity[$i]->$eachSetter($convertedData[$caseImport]);
+                            }
+                            $j++;
+                            $caseImport++;
                         }
-                        $j++;
-                        $caseImport++;
+                        $j = 0;
+                        $i++;
                     }
-                    $j = 0;
-                    $i++;
-                }
-                //csv reading end
-                // add Links for each entities
-                $k = 0;
-                $slug = $this->slugificator->slugFactory($row[0]);
+                    //csv reading end
+                    // add Links for each entities
+                    $k = 0;
+                    $slug = $this->slugificator->slugFactory($row[0]);
 
-                foreach ($softEntitiesYml as $entity) {
+                    foreach ($softEntitiesYml as $entity) {
 
-                    //ATTENTION: Rajouter une boucle si les entités ont plusieurs links et plusieurs sources
+                        //ATTENTION: Rajouter une boucle si les entités ont plusieurs links et plusieurs sources
 
-                    if ($entity["links"]["relation"] === "Many-to-Many") {
-                        $eachSetterLink = "add" . $entityKeys[$k];
-                        $eachSource = "AppBundle\\Entity\\" . $entity["links"]["source"];
-                        $eachSource->$eachSetterLink($eachEntity[$k]);
-                    }
-                    //upgrade: $eachEntity[0] can be change by an automatic
-                    if ($entity["links"]["relation"] === "One-to-One") {
-                        $eachSetterLink = "set" . $entityKeys[$k];
-                        $eachEntity[0]->$eachSetterLink($eachEntity[$k]);
-                    }
+                        if ($entity["links"]["relation"] === "Many-to-Many") {
```

Comment:
que se passe-t-il si on renseigne un type de liaison non géré (par exemple "'many to many" ?

Reply:
Rien, on l'a pas géré car on en a pas besoin, mais on en avait parlé et on avait convenu qu'on gardait ces améliorations là pour la v2